### PR TITLE
[Fix] New Reddit API and Error handler

### PR DESCRIPTION
**Pull request by alvinbengeorge**
# Description

- `meme-api` died because of heroku free plan issues. I switched jack over to a new API, you can find the repository's link [here](https://github.com/alvinbengeorge/redditAPI).
- An unknown issue made jack crash which made it restart couple of times, error handler to fix that issue

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Screenshot
image

### CODE REVIEW

**Comment by shawshankkumar** [comment]
@alvinbengeorge You can work on the commit names (bit too long)

File: src/helper/kzillaXYZ.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export async function handleShrinkURLMessage(
         createBasicEmbed(ERRORS.URL_MISSING, "ERROR")
       );
     }
-  } catch (err) {
+  } catch (err: any) {
```
Comment (shawshankkumar) [inline src/helper/kzillaXYZ.ts:51]: Why any?

File: src/utils/constants.ts [hunk 2]
```diff
@@ -445,7 +445,7 @@ export const CONSTANTS = {
   KZILLA_XYZ_SHRINK_URL_ENDPOINT: "https://kzilla.xyz/api/v1/webhook/link",
   JOKES_URL_ENDPOINT:
     "https://official-joke-api.appspot.com/jokes/programming/random",
-  MEMES_API: "https://meme-api.herokuapp.com/gimme/",
+  MEMES_API: "https://redditAPI.alvinbengeorge.repl.co/meme/", // change it to a new one if required, source code is available at https://github.com/alvinbengeorge/redditAPI
```
Comment (shawshankkumar) [inline src/utils/constants.ts:448]: attach screenshots of the working command please

File: src/helper/kzillaXYZ.ts (hunk 1 above)
Comment (alvinbengeorge) [reply src/helper/kzillaXYZ.ts:51]: Oh no, changing it to `: Error`

Comment (alvinbengeorge) [reply src/helper/kzillaXYZ.ts:51]: I'm pretty sure most of the places you had error as `any` datatype, I must've gone with that at the moment, anyway, changing that ASAP

File: src/utils/constants.ts (hunk 2 above)
Comment (alvinbengeorge) [reply src/utils/constants.ts:448]: Edited the primary message

**Review by shawshankkumar — COMMENTED** [review]
@sm-sami @Developer-R-7 Review

**Pull request closed without merging**